### PR TITLE
User login and logout functionality

### PR DESCRIPTION
**Pull request by CoopTang**
- (#29 ) Users cannot log in with invalid credentials
- (#28 ) Users that are logged in are redirected to their appropriate pages
- (#27 ) Users can log out, with their cart cleared, and redirected to home page

### CODE REVIEW

File: spec/features/users/navigation_spec.rb [hunk 1]
```diff
@@ -36,7 +36,7 @@
         click_link 'Log Out'
       end
 
-      expect(current_path).to eq('/logout')
+      expect(current_path).to eq(welcome_path)
```
Comment (lrs8810) [inline spec/features/users/navigation_spec.rb:39]: Thanks for using the path helper!

File: spec/features/users/user_logout_spec.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        email: 'bob@email.com',
+        password: 'secure'
+      )
+
+      fill_in :email, with: user.email
+      fill_in :password, with: user.password
+
+      click_on 'Sign me in'
+
+      meg = Merchant.create(name: "Meg's Bike Shop", address: '123 Bike Rd.', city: 'Denver', state: 'CO', zip: 80_203)
```
Comment (lrs8810) [inline spec/features/users/user_logout_spec.rb:23]: This is a great case for implementing factory bot.

**Review by lrs8810 — APPROVED** [review]

**Review by cjkelling — APPROVED** [review]
These changes look good. Your testing is set up well and shows that your functionality is working.

**Comment by freeheeling** [comment]
Wonderful job driving.

**Pull request merged**